src/plugins/dashboard/server/routes/create.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { VersionedRouter } from '@kbn/core-http-server';
+import { LATEST_VERSION } from '@kbn/data-views-plugin/common';
+import { v2023_10_31 } from '../../common/api';
+import { DashboardStorage } from '../content_management';
+
+export function registerCreate(router: VersionedRouter, cmStorage: DashboardStorage) {
+  router
+    .post({
+      path: '/api/dashboard',
+      access: 'public',
+      description: 'Create a dashboard',
+    })
+    .addVersion(
+      {
+        version: '2023-10-31',
+        validate: {
+          request: {
+            body: v2023_10_31.dashboardCreate,
+          },
+          response: {
+            200: { body: v2023_10_31.dashboard },
+          },
+        },
+      },
+      async (ctx, req, res) => {
+        const { id: reqId, ...attrs } = req.body;
+        let id = reqId;
+        try {
+          ({
+            item: { id },
+          } = await cmStorage.create(
+            {} as any,
```

Comment:
Answering your question from the demo, I think a better approach would be to use here server-side content management client instead of a dashboard storage directly https://github.com/elastic/kibana/pull/175968

But this doesn't change the general idea. great demo!

Reply:
Thanks, I did not know about that, I've updated the code to use `getForRequest`. Passing values into `create` from the client seems to not have type checking working, but otherwise code interfaces work as expected (did not actually test 😅 ).